tests/manage/pv_services/test_resource_deletion_during_pvc_pod_creation_and_io.py

```diff
@@ -0,0 +1,275 @@
+        creation and IO operation are progressing.
+        Verifies PVCs can be re used by creating new pods.
+
+        Steps:
+        1. Create pods for running IO and verify they are Running.
+        2. Start creating more pods.
+        3. Start creating new PVCs.
+        4. Start IO on pods created in Step 1.
+        5. Delete the resource 'resource_to_delete'.
+        6. Verify that PVCs created in Step 3 are in Bound state.
+        7. Verify that pods created in Step 2 are Running.
+        8. Verify IO results.
+        9. Delete pods created in Steps 1 and 2.
+        10. Use all PVCs to create new pods. One PVC for one pod.
+        11. Start IO on all pods created in Step 10.
+        12. Verify IO results.
+        """
+        # Separate the available PVCs
+        pvc_objs_for_io_pods = self.pvc_objs[0:self.pvc_num_for_io_pods]
+        pvc_objs_new_pods = self.pvc_objs[self.pvc_num_for_io_pods:]
+
+        executor = ThreadPoolExecutor(max_workers=2)
+
+        DISRUPTION_OPS.set_resource(resource=resource_to_delete)
+
+        # Create pods for running IO
+        io_pods = helpers.create_pods(
+            pvc_objs_list=pvc_objs_for_io_pods, interface_type=self.interface,
+            desired_status=constants.STATUS_RUNNING, wait=True,
+            namespace=self.namespace
+        )
+
+        # Updating self.pod_objs for the purpose of teardown
+        self.pod_objs.extend(io_pods)
+
+        # Do setup for running IO on pods
+        log.info("Setting up pods for running IO")
+        for pod_obj in io_pods:
+            pod_obj.workload_setup(storage_type='fs')
+        log.info("Setup for running IO is completed on pods")
+
+        # Start creating new pods
+        log.info("Start creating new pods.")
+        bulk_pod_create = executor.submit(
+            helpers.create_pods, pvc_objs_list=pvc_objs_new_pods,
+            interface_type=self.interface, wait=False,
+            namespace=self.namespace
+        )
+
+        # Start creation of new PVCs
+        log.info("Start creating new PVCs.")
+        bulk_pvc_create = executor.submit(
+            helpers.create_multiple_pvcs, sc_name=self.sc_obj.name,
+            namespace=self.namespace, number_of_pvc=self.num_of_new_pvcs,
+            size=self.pvc_size
+        )
+
+        # Start IO on each pod
+        log.info("Start IO on pods")
+        for pod_obj in io_pods:
+            pod_obj.run_io(storage_type='fs', size=f'{self.pvc_size_int - 1}G')
+        log.info("IO started on all pods.")
+
+        # Delete the resource
+        DISRUPTION_OPS.delete_resource()
+
+        # Getting result of PVC creation as list of PVC objects
+        pvc_objs_new = bulk_pvc_create.result()
+
+        # Updating self.pvc_objs_new for the purpose of teardown
+        self.pvc_objs_new.extend(pvc_objs_new)
+
+        # Verification of PVC status is done in 'create_multiple_pvcs' function
+        log.info("Verified: New PVCs are Bound.")
```

Comment:
let's ensure this is happening here instead of somewhere else.

Reply:
Verification is added in latest commit.

# Verify PVCs are Bound
        for pvc_obj in pvc_objs_new:
            assert pvc_obj.ocp.wait_for_resource(
                condition=constants.STATUS_BOUND, resource_name=pvc_obj.name,
                timeout=240, sleep=10
            ), (
                f"Wait timeout: PVC {pvc_obj.name} is not in 'Bound' status"
            )
        log.info("Verified: New PVCs are Bound.")

Reply:
Verification is added in latest commit.
```
        # Verify PVCs are Bound
        for pvc_obj in pvc_objs_new:
            assert pvc_obj.ocp.wait_for_resource(
                condition=constants.STATUS_BOUND, resource_name=pvc_obj.name,
                timeout=240, sleep=10
            ), (
                f"Wait timeout: PVC {pvc_obj.name} is not in 'Bound' status"
            )
        log.info("Verified: New PVCs are Bound.")
```